src/auth/index.js

```diff
@@ -274,4 +274,65 @@ export default class Auth {
     this.storage.remove(authCommon.IMPERSONATION_USER_KEY);
     this.storage.remove(authCommon.IMPERSONATION_REAL_USER_AUTH_KEY);
   }
+
+  parseRedirectFragment(fragment, ourState) {
+    // After being redirected from oauth, the URL will look like:
+    // https://todo.examples.stitch.mongodb.com/#_stitch_state=...&_stitch_ua=...
+    // This function parses out stitch-specific tokens from the fragment and
+    // builds an object describing the result.
+    const vars = fragment.split('&');
+    const result = { ua: null, found: false, stateValid: false, lastError: null };
+    let shouldBreak = false;
+    for (let i = 0; i < vars.length; ++i) {
+      const pairParts = vars[i].split('=');
+      const pairKey = decodeURIComponent(pairParts[0]);
+      switch (pairKey) {
+      case authCommon.STITCH_ERROR_KEY:
+        result.lastError = decodeURIComponent(pairParts[1]);
```

Comment:
[semi-nit] I think i would prefer making another const for this `pairPart` just to be more descriptive.